docs/getting_started_jax.ipynb

```diff
@@ -0,0 +1,669 @@
+          ]
+        }
+      ],
+      "source": [
+        "!pip install jax==0.4.35"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "id": "4flIPFVmAYNw"
+      },
+      "outputs": [],
+      "source": [
+        "#@title Imports and utilities\n",
+        "\n",
+        "import os\n",
+        "# make sure our code runs on 8 devices\n",
+        "os.environ['XLA_FLAGS'] = '--xla_force_host_platform_device_count=8'\n",
+        "\n",
+        "import jax\n",
+        "import numpy as np\n",
+        "from jax import numpy as jnp\n",
+        "from jax.sharding import Mesh, NamedSharding, PartitionSpec\n",
+        "from jax._src.sharding_impls import UNSPECIFIED\n",
+        "from jax.experimental.shard_map import shard_map"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "fl3cXGo9CRMo"
+      },
+      "source": [
+        "First, let's create our mesh."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 3,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "9NTq9HXCE5OQ",
+        "outputId": "99546727-22d5-4a46-bb6f-09b01094b4b8"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "OrderedDict([('data', 4), ('model', 2)])\n"
+          ]
+        }
+      ],
+      "source": [
+        "mesh = Mesh(\n",
+        "    np.reshape(np.array(jax.devices()), (4, 2)),\n",
+        "    ('data', 'model'))\n",
+        "\n",
+        "print(mesh.shape)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "YjgahNBpNu19"
+      },
+      "source": [
+        "### In/Out shardings\n",
+        "\n",
+        "Let's look at what changed the most: how sharding attributes are encoded in the JAX program for the compiler to read.\n",
+        "\n",
+        "Let's look at it through an example. It's going to be an MLP-like model consisting of no bias tensors, and 2 layers (two matmuls)."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 4,
+      "metadata": {
+        "id": "A--HECIHAe5E"
+      },
+      "outputs": [],
+      "source": [
+        "def predict(x, w1, w2):\n",
+        "  x = jnp.tanh(x)\n",
+        "  z1 = jnp.einsum('ij,jk->ik', x, w1)\n",
+        "  z2 = jnp.einsum('ij,jk->ik', z1, w2)\n",
+        "  return jnp.sin(z2)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "kLR4IUmXGb7b"
+      },
+      "source": [
+        "What we will want to do here sharding wise is:\n",
+        "\n",
+        "1. `data` parallelism on x\n",
+        "2. `model` parallelism on `w1` and `w2` through the [megatron](http://go/arxiv/1909.08053) sharding strategy.\n",
```

Comment:
Didn't realize this until my external talk, but better to cal this `tensor` parallelism instead of `model` parallelism.